src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -1168,5 +1166,20 @@ ResourceSet ClusterTaskManager::CalcNormalTaskResources() const {
   return total_normal_task_resources;
 }
 
+void ClusterTaskManager::ForAllQueues(std::function<void(const Work &)> &fn) const {
```

Comment:
This function looks a little bit wired actually. Basically, it wants to iterate all workers in this queue and we have multiple queues there.  Usually boost.range.join will help to chain containers, but here it's nested with chain. I guess it's fine now.

Reply:
Yeah... `boost::join` doesn't seem to work because the types of the iterators are different now. Maybe this is a bit of a noob c++ question, but pls let me know if there's a way to use `boost::join` still.